src/main/drivers/accgyro/accgyro_spi_bmi270_config.h

```diff
@@ -0,0 +1,493 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+* Copyright (c) 2020 Bosch Sensortec GmbH. All rights reserved.
+*
+* BSD-3-Clause
```

Comment:
Is that ok file already under the BSD-3 license be released under the GPL？

Reply:
That's something we still need to discuss. The only term I see as a problem is:
```
2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
```
So as I read this we could possibly just include the required info as a static that gets compiled in but not used for anything. This would meet the terms of having it "included" in the binary distribution. It's a waste of space, but we already can't include this driver in any target without 1MB flash anyway.

Reply:
N.b.: All of the STM libraries that are used in Betaflight are already BSD-3, so this is not much different. And all we have in the STM libraries is a comment in the file header linking to an URL for the license.

```
Redistributions in binary form must reproduce the above copyright notice [...] in the documentation and/or other materials provided with the distribution.
```

To my understanding this does not mean at all that the license must be included _in the binary_ - having it as part of the documentation (which includes all of the repository) will be good enough, so sticking a license file into the directory containing the microcode should be sufficient to meet the requirements.

Reply:
After looking through the code now, I think what we should be doing is to just plop the `bmi270.[ch]` and `LICENSE` files from https://github.com/BoschSensortec/BMI270-Sensor-API/ into `lib/main/BoschSensortec` or similar, and reference the microcode from there - this will solve the licensing issue, and also minimise the amount of work required if the microcode ever gets updated.

Reply:
I don't have a problem putting the config, license, etc. in the lib directories. But not sure we want to literally use their `bmi270.c` and `bmi270.h`. They have a bunch of stuff we don't want to compile in - some functions and a whole bunch of defines we don't need. The `accgyro_spi_bmi270_config.h` file I made was basically the `bmi270.c` file with the extra stuff removed.

Reply:
I don't think this should make a difference:
- the library header should only be included in `accgyro_spi_bmi270_config.h`, so any defines will not be exposed outside of that, which I think is acceptable;
- nothing in the library except for the array containing the microcode will be referenced anywhere in the firmware, so the linker will eliminate it - this is already the case for all other libraries (including the massive STM libraries), so this should be fine.